terraform/modules/data-share-service/lb-security.tf

```diff
@@ -1,41 +1,35 @@
-# Auto-updating security group for Cloudfront
-# See https://aws.amazon.com/blogs/security/how-to-automatically-update-your-security-groups-for-amazon-cloudfront-and-aws-waf-by-using-aws-lambda/
-locals {
-  lb_sg_protocols = ["http"]
-  lb_sg_names     = ["cloudfront_g", "cloudfront_r"]
-
-  lb_sg_opts = setproduct(local.lb_sg_names, local.lb_sg_protocols)
-
-  # Central AWS IP address changes topic
-  # See https://aws.amazon.com/blogs/aws/subscribe-to-aws-public-ip-address-changes-via-amazon-sns/
-  update_ips_sns_arn = "arn:aws:sns:us-east-1:806199016981:AmazonIpSpaceChanged"
-}
-
-resource "aws_security_group" "lb_auto" {
-  count       = length(local.lb_sg_opts)
-  name_prefix = "${var.environment}-ecs-alb-auto-${count.index}-"
-  description = "${local.lb_sg_opts[count.index][1]} access to GDX data share POC LB from Cloudfront for ${local.lb_sg_opts[count.index][0]}"
+resource "aws_security_group" "lb_cloudfront" {
+  name        = "${var.environment}-lb-cloudfront"
+  description = "Allow access to GDX data share POC LB from Cloudfront"
   vpc_id      = module.vpc.vpc_id
 
-  egress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = [var.vpc_cidr]
-    description = "LB egress rule"
-  }
-
   lifecycle {
-    # Ingress will be set automatically via a Lambda
-    ignore_changes        = [ingress]
     create_before_destroy = true
   }
+}
 
-  tags = {
-    Name       = local.lb_sg_opts[count.index][0],
-    AutoUpdate = "true",
-    Protocol   = local.lb_sg_opts[count.index][1]
-  }
+data "aws_ec2_managed_prefix_list" "cloudfront" {
+  name = "com.amazonaws.global.cloudfront.origin-facing"
+}
+
+resource "aws_security_group_rule" "lb_cloudfront" {
+  type              = "ingress"
+  protocol          = "tcp"
+  from_port         = 80
+  to_port           = 80
+  description       = "LB ingress rule for cloudfront"
+  security_group_id = aws_security_group.lb_cloudfront.id
+  prefix_list_ids   = [data.aws_ec2_managed_prefix_list.cloudfront.id]
+}
+
+resource "aws_security_group_rule" "lb_egress" {
+  type              = "egress"
+  protocol          = "-1"
+  from_port         = 0
+  to_port           = 0
```

Comment:
Is there any reason this isn't more specific?

Reply:
This is to make sure there are no extra changes here